tasks/release-git-rebase.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: release-git-rebase
+spec:
+  resources:
+    inputs:
+      - name: git-source-repository
+        type: git
+  params:
+    - name: upstream-url
+      description: URL pointing to upstream repository
+      default: https://github.com/4n4nd/pipeline-test-upstream.git
+    - name: pull-api
+      description: pull api url
+      default: https://api.github.com/repos/4n4nd/pipeline-test-upstream/pulls
+  steps:
+    - name: rebase
+      workingDir: /workspace/git-source-repository
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: auth-secret
+              key: password
+        - name: GITHUB_USERNAME
+          valueFrom:
+            secretKeyRef:
+              name: auth-secret
+              key: username
+        - name: GITHUB_EMAIL
+          valueFrom:
+            secretKeyRef:
+              name: auth-secret
+              key: user-email
+      image: mrnonz/alpine-git-curl:latest
+      script: |
+        git config --local user.email "$GITHUB_EMAIL"
+        git config --local user.name "$GITHUB_USERNAME"
+        git remote add upstream "$(params.upstream-url)"
+        git fetch upstream
+        git checkout -b branch_rebasing
+        git rebase upstream/master
+        git push origin branch_rebasing
```

Comment:
The service account which is link to auth secret would only help in authenticating ssh URL which is used in process of cloning the repo to a directory.

for this operation, we would need to mount the ssh key into the directory.